### PR TITLE
Enable telescoping nests (multiple levels)

### PR DESCRIPTION
**Pull request by JosephMouallem**
These modifications will allow multiple telescoping nests functionality.
Telescoping nests are defined as nests within nests.
This works for many nest levels.

### CODE REVIEW

**Review by bensonr — APPROVED** [review]

**Comment by ngs333** [comment]
@JosephMouallem , 
I am approving this PR, but do you think a unit test (directoryy  FRE-NCtools/t ) should be added or modified to cover the changes you introduced?

**Review by ngs333 — APPROVED** [review]

**Comment by ngs333** [comment]
@JosephMouallem Do you already have the tests? If so, do you need help in adding them?

**Review by ceblanton — APPROVED** [review]
Looks good, and agree a test should be added

**Comment by ngs333** [comment]
@JosephMouallem 
Joseph, just letting you know I'll wait for the unit tests prior to merging this PR.
There was a discussion on whether to merge now and latter merge the unit test under a different github issue,
but it was decided to hold the merge.  Thanks

**Comment by JosephMouallem** [comment]
@ngs333 Hi Miguel,
I added a file that tests for the generation of same level and telescoping nests, let me know if this works.

File: t/Test27-multiple_nests.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#Make_hgrid: create three same level -level1- nests in tiles 2,5,6"
+  run command make_hgrid \
+		--grid_type gnomonic_ed \
+		--nlon 96 \
+		--grid_name C48_grid \
+		--do_schmidt \
+		--stretch_factor 1.0 \
+		--target_lon -97.5 \
+		--target_lat 36.5 \
+		--nest_grids 3 \
```
Comment (ceblanton) [inline t/Test27-multiple_nests.sh:43]: Hi @JosephMouallem `--nest_grids` should be singular, `--nest_grid` here and below in line 66

**Comment by ceblanton** [comment]
Hi @JosephMouallem could you please merge your branch with the updated master, and push back? We have fixed a failing test recently that should allow your tests to pass on the github CI (also, the make_grid option `--nest_grids` should be `--nest_grid`)

**Comment by ngs333** [comment]
@JosephMouallem 
I think the only thing missing is to add the name of the test shell script to the Makefile.am in directory t.
If you do that then we can automatically run it in as one of the tests in the suite of unit tests. For example, if you run "make check" after you run "make", then you will see  all the unit tests run, including yours. Its just one line to add, once you are done, the middle of Makefile.am should look like this:

        Test26-reference_fregrid_gcc.sh \
	Test27-multiple_nests.sh \
        01-make_hgrid.sh \
        01-make_vgrid.sh

**Review by ngs333 — CHANGES_REQUESTED** [review]
See comment on Makefile.am.

**Comment by JosephMouallem** [comment]
@ceblanton Hi Chris, I don't think --nest_grid is working anymore (the code doesn't fail but it only creates one additional grid if used). To create the multiple nests --nest_grids should be used with an argument (number of nests). If this creates issues in the future, we can add a reference directory to make sure we are doing the right thing.

@ngs333 Miguel, please see the updated Makefile.am, I think it is working correctly.

Thanks!

**Review by ngs333 — APPROVED** [review]

**Pull request merged**